src/Auth0/Login/Auth0Service.php

```diff
@@ -167,7 +173,28 @@ public function rememberUser($value = null)
      */
     public function decodeJWT($encUser, array $verifierOptions = [])
     {
-        $this->apiuser = $this->auth0->decodeIdToken($encUser, $verifierOptions);
+        $token_issuer = 'https://'.$this->auth0Config['domain'].'/';
+        $apiIdentifier = $this->auth0Config['api_identifier'];
+        $idTokenAlg = $this->auth0Config['supported_algs'][0] ?? 'RS256';
+
+        $signature_verifier = null;
+        if ('RS256' === $idTokenAlg) {
+            $jwksUri = $this->auth0Config['jwks_uri'] ?? 'https://'.$this->auth0Config['domain'].'/.well-known/jwks.json';
+            $jwks_fetcher = new JWKFetcher($this->auth0Config['cache_handler']);
+            $jwks = $jwks_fetcher->getKeys($jwksUri);
+            $signature_verifier = new AsymmetricVerifier($jwks);
+        } else if ('HS256' === $idTokenAlg) {
+            $signature_verifier = new SymmetricVerifier($this->auth0Config['client_secret']);
+        }
```

Comment:
else... alg not supported? if this is not possible at this point, please drop a comment there for the next maintainer / security reviewer

Reply:
This came up in the PHP SDK as well. It does look a bit like `null` passed to `TokenVerifier` could be treated as `alg:none` (which is not the case but still). Same in the PHP SDK:

https://github.com/auth0/auth0-PHP/blob/master/src/Auth0.php#L657

I would throw an `InvalidTokenException` here.

https://github.com/auth0/auth0-PHP/blob/master/src/Exception/InvalidTokenException.php

Reply:
👍Pushed another commit that throws an `InvalidTokenException` when using an unsupported signing algorithm.